pkg/feeds/artifactory_generic.go

```diff
@@ -0,0 +1,43 @@
+package feeds
+
+import (
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/internal"
+	"github.com/go-playground/validator/v10"
+	"github.com/go-playground/validator/v10/non-standard/validators"
+)
+
+type ArtifactoryGenericFeed struct {
+	LayoutRegex string `json:"LayoutRegex,omitempty"`
+	FeedURI     string `json:"FeedUri,omitempty"`
+	Repository  string `json:"Repository"`
+
+	feed
+}
+
+func NewArtifactoryGenericFeed(name string) (*ArtifactoryGenericFeed, error) {
+	if internal.IsEmpty(name) {
+		return nil, internal.CreateRequiredParameterIsEmptyOrNilError("name")
```

Comment:
Avoid IsEmpyOrNilError for string, trying to change it to Is `CreateRequiredParameterIsEmptyError` as strings can't be nil. Only really a nitpick.
```suggestion
		return nil, internal.CreateRequiredParameterIsEmptyError("name")
```